components/eamxx/src/physics/shoc/atmosphere_macrophysics.cpp

```diff
@@ -461,7 +461,7 @@ void SHOCMacrophysics::run_impl (const int dt)
   // number of SHOC timesteps (nadv) to be 1.
   // TODO: input parameter?
   hdtime = dt;
-  m_nadv = std::max(hdtime/dt,1);
+  m_nadv = std::max(static_cast<int>(hdtime/dt),1);
```

Comment:
Need to be careful here. Casting to an int will always round down. `round` should be used for proper rounding.

Reply:
That said, `hdtime = dt` means this will always be 1, anyway.

Reply:
Good point. We don't want to round 2.9999 to 2 if/when we will allow hdtime!=dt. I can fix this.